CommunicationServer/Application/Entities/ChessTimer.cs

```diff
@@ -0,0 +1,93 @@
+using Application.Utils;
+using Domain.DTOs;
+using Rudzoft.ChessLib.Enums;
+
+namespace Application.Entities;
+
+public class ChessTimer
+{
+    private readonly PausableTimer _whiteTimer = new(1000.0);
+    private readonly PausableTimer _blackTimer = new(1000.0);
+    private readonly uint _timeControlSeconds;
+    private readonly uint _timeControlIncrement;
+    private readonly ValueWrapper<bool> _whitePlaying;
+    public double WhiteRemainingTimeMs { get; private set; }
+    public double BlackRemainingTimeMs { get; private set; }
+    
+    public delegate void EventHandler(object sender, EventArgs args, ResponseJoinedGameDto dto);
+    public event EventHandler ThrowEvent = delegate{};
+
+    public ChessTimer(ref ValueWrapper<bool> whitePlaying, uint timeControlSeconds, uint timeControlIncrement)
+    {
+        _whitePlaying = whitePlaying;
+        
+        _timeControlSeconds = timeControlSeconds;
+        _timeControlIncrement = timeControlIncrement;
+
+        WhiteRemainingTimeMs = timeControlSeconds * 1000;
+        BlackRemainingTimeMs = timeControlSeconds * 1000;
+        
+        _whiteTimer.Elapsed += OnOneSecElapsed;
+        _blackTimer.Elapsed += OnOneSecElapsed;
+    }
+
+    public void StartTimers()
+    {
+        _whiteTimer.Start();
+        _blackTimer.Start();
+    }
+    
+    public void UpdateTimers()
+    {
+        if (_whitePlaying.Value)
+        {
+            _whitePlaying.Value = false;
+            WhiteRemainingTimeMs += _timeControlIncrement * 1000;
```

Comment:
Maybe store the increment in ms as you only use it that way

Reply:
_timeControlIncrement is a values received from the user when choosing the time control, which is in seconds. I suppose I could change it to ms

Reply:
Ooor You could multiple it by a 1000 in a constructor

Reply:
Will do